app/models/api_request.rb

```diff
@@ -0,0 +1,47 @@
+class ApiRequest
+  def query(query)
+    uri = request_uri(query)
+    res = send_api_request(uri)
+    format_response(res)
+  end
+
+  def request_uri(query = nil)
+    URI::HTTPS.build(
+      host: host,
+      path: path,
+      query: query
```

Comment:
this method won't need a param (it will be coming from the class itself) if you take my above suggestion!